### DISCUSSION TITLE
Refactor baseline minting to "use it or lose it"

### DISCUSSION THREAD

**Discussion by jnthnvctr** [filecoin-project/FIPs#887] [Enhancements - Misc]
# Motivation
_Note: This one part of a three part proposal to help accelerate growth in the ecosystem - but the only one that requires a FIP. Broadly speaking the three parts are: (a) tighten up Filecoin's tokenomics (b) align lenders around a notion of a "base rate" tied to FIL inflation and (c) accelerate paid services on the network (e.g. paid storage on ramps, integrations with compute protocols, etc)._

- Filecoin’s tokenomics are broadly misunderstood
  - It is hard to talk about Filecoin’s max supply given:
     - Original tokenomics designs talk about 2b as the total supply and 
     -  That 2b number is only applicable if certain conditions were to be true (in actuality we’ve already burned several million tokens, to release the mining reserve a FIP would need to be implemented, and of course the network would have to grow to certain sizes)
   - Baseline minting’s unique tie to hardware capacity (vs time) adds complexity for analyzing Filecoin against other ecosystems (especially given FDV is a metric that is commonly used).
- This has a real cost to the Filecoin economy.
  - Filecoin is evaluated against a “worst case” scenario for inflation - one that assumes specific conditions (20 years pass, 1YiB of raw capacity acquired, FIP for the mining reserve) that may never come to be. 
     - **_Effectively the network is paying a tax today based on a future state of the network that may never actually exist_.**
   - Baseline minting’s deferred inflation model creates an overhang of potential future dilution
      - This future dilution aims to rewards participants who may not exist with rewards that may not be released at the expense of builders and SPs, today.
      - This construction fails to account for the path dependency of economic growth, and that Filecoin does not exist in a vacuum => we want capital inflows to support activity today so we can accelerate the development of ongoing infrastructure.
- This cost directly constricts the network’s economic growth
    - This translates to a lower TAM for any builder that wants to target Filecoin as a prospective target to build today, making it harder to attract talent and accelerate development to fill important needs in the ecosystem.
    - This also means the revenues of SPs today (at least partially denominated in FIL) are unfairly discounted
    - This also makes it (relatively) more expensive for the ecosystem to invest in its public goods
    - This makes it (relatively) less attractive for other protocols’ to prioritize FVM vs other opportunities (lower TAM)

## Proposal
- We change baseline minting to become “use it or lose it” 
    - Practically this means whenever Filecoin dips below baseline (as it has since Feb ‘23) - the _**unreleased**_ baseline minting rewards are removed from the total supply (i.e. inflation is no longer deferred; we can classify this as burn or just decrement the total supply).
       - This rewards existing FIL holders (SPs, tokenholders, developers) equally by giving them a higher proportionate share of the (theoretical long term) total supply by continuing to participate in the network
       - It further increases the incentive for SPs to grow hardware even while others hold off, and improves the incentive for tokenholders to lend to support growth
           - Consideration: Impact on SP incentives to grow or not grow
              - This should not change the FIL-on-FIL game theory - if anything, this should increase the attractiveness to grow while others sit on the sidelines because SPs have more assurances that they will capture a higher portion of the network when baseline minting is low. 
            - Consideration: Impact on tokenholder incentives to lend or not lend
              - Given there is still inflation slated at least until 2040 via simple minting, this proposal should not change the tokenholder incentives to lend or not lend
              - If anything, long term aligned tokenholders now have an juiced incentive for continuing to take long duration risk on FIL by continuing to lend and support SP growth during down markets
              - Reducing the total supply effectively rewards long term aligned holders with a greater share of the network - and lending out your tokens to support growth allows you to capture a share of the remaining inflation.
            - Consideration: DeFi and other internal FVM economies
              - This shouldn't affect anything internal to the Filecoin economy as it only affects block rewards that would've been deferred
- For the purposes of this proposal, we align the “timeline” for baseline minting to that of simple minting to simplify understanding. Practically this means that: 
    - All deferred inflation since Feb ‘23 should be removed permanently from total supply.
       - (I've attempted to quantify this, but my best estimate is eyeballing this [chart](https://filecoin.io/blog/filecoin-circulating-supply/) at 20-30mm of unreleased FIL. Prior to a FIP we should clarify the total impact here).
- While network power remains below baseline, the max theoretical supply should continue to shrink.
- This proposal should not affect any anticipated FIL-on-FIL returns, simply make FIL a scarcer asset overall, and increase the incentives to return to baseline growth

**Related notes:**
- The role of inflation is an active topic in other ecosystems, and has been received [well](https://x.com/cosmoshub/status/1728519681236177234?s=46&t=Ab0Q1RGim72hU6oj8LygCw)
  - As some have pointed out in draft discussions, circulating supply is a first order driver - but FDV is increasingly used as a comparison metric (can share more examples if desired, but here's a [sample](https://6thman.ventures/writing/how-depin-improves-infrastructure-systems/) from a sophisticated firm) and has a meaningful effect when people evaluate ecosystems for growth potential.
     - It's important to contextualize this in terms of opportunity cost - if we do not add a time dimension to baseline minting, the rational choice is to deploy resources elsewhere (short term XRP trade, or putting capital against the Bitcoin halvening) because future rewards are always on the table. 
     - This consequently slows the rate at which investments that can have compounding effects on the network happen 
- This proposal is one of several areas of improvement that can drive long term success for the ecosystem. It isn’t a substitute for the ongoing investments in increasing paid offerings into the networks’ hardware operators (e.g. Banyan, Lighthouses, Seals, Steeldomes, etc), or improving the robustness of the on-chain offerings (e.g. more sophisticated DeFi tooling)

(Thank you to the GLIF, Banyan, and the others who helped shape this proposal and gave early feedback)

**Comment by laudiacay** [comment]
banyan bless 🌳

**Comment by whyrusleeping** [comment]
This seems like a great idea to me 👍

**Reply by Bilal9890** [reply]
Great

**Comment by Fatman13** [comment]
I would highly suggest others, in the pie chart below, to "lose it" too so that it is fair to the initial design of FIL distribution.

Or we can simply burn the 15% mining reserve and the 2.5% fundraising remainder to achieve similar end result.

![image](https://github.com/filecoin-project/FIPs/assets/1591330/aaf18096-2ce1-46ab-a014-68098dc0eef8)

**Reply by jnthnvctr** [reply]
That isn't the initial proportion - that's a proportion that exists only under certain conditions. Baseline minting is not released unless specific conditions are met and the mining reserve is not automatically released. 

This proposal doesn't confiscate anyone's earned resources, but encourages us to get back to baseline faster (and aims to make it easier for new start ups to raise to build the infrastructure that can help grow the FIL economy overall)

If we move back to growth, the burn stops and your chart is close to what the final distribution looks like. By creating time urgency we can incent more people to put FIL to work sooner to support that growth.

**Reply by Fatman13** [reply]
> That isn't the initial proportion - that's a proportion that exists only under certain conditions.

I meant the distribution of FIL dictated by the [white paper](https://filecoin.io/2020-engineering-filecoins-economy-en.pdf). If portions of 55% mining allocation are to be burned by any means other than the initial design (EIP-1559 etc), then it would only be fair that each other parties in the FIL distribution burns their own shares so that the initial FIL distribution proportion remains. 

For example, if after "use it or lose it", mining allocation becomes 30 (25 burned), then FF needs to burn 2.3 and come down to 2.7, PL team and contributor needs to burn 2.1 and come down to 2.4, PL needs to burn 4.8 and come down to 5.7, fundraising remainder needs to burn 1.2 and come down to 1.3, mining reserve needs to burn 6.9 and come down to 8.1.

Or like I said we could burn 15% mining reserve and the 2.5% fundraising remainder without changing protocol and is fair to most network participants.

**Reply by jnthnvctr** [reply]
This logic doesn't make sense to me. 

PL and FF's allocation was for launching the network - and they linearly vests for 6 years, similar to the SAFT investors who put up capital. Surely the network could have been designed to say that they instantly vest at day 1, but that obviously would not be good for long term alignment. 

Your argument seems to be that we should confiscate someone else's earned portion to keep it in proportion with unearned rewards that might not be released (e.g. network never grows past XX EiB). Note that under the current model, deferred baseline rewards means PL / FF / early VC allocations are a higher portion of current circulating supply - which if lent out will capture the imaginary baseline rewards if we ever go back to baseline. 

Burning deferred rewards isn't affecting anyone's cashflow, it isn't even definitively affecting the final distribution - but it is affecting an incentive to put capital / resources into the network today. On reflection, I'm actually not sure a deferred inflation model ever really made sense.

Burning the mining reserve / fundraising allocation doesn't actually solve for the time dimension that I laid out above. In an earlier iteration I'd raised something similar - the flag I'd received was that this may trigger larger debates given FIPing out things is a 1 way door vs a 2 way door. The argument I sort of buy here is that it gives the ecosystem leeway to FIP in more inflation in the future if needed (say after 2040 if theres a necessity to have a slow dribble of inflation). Imho what matters here is shipping something that creates an opportunity cost to incent deploying resources into the network vs elsewhere.

This proposal is a 2 way door - if it instigates growth soon after, the net burn is minimal and we're still in the same world we were in before. But in the interim, it would allow folks to credibly point out what the total supply in 2040 would be on the network's current trajectory (which given the magnitude of baseline minting is dramatically lower than the current market estimates). 

To underscore on the time dimension: Filecoin needs to become a currency you can save in - without having a reason to deploy capital into the network today (and having an overhang that can be captured at some future date) - you're effectively creating an incentive to _not_ save in FIL. Any FIP to add more inflation from reserves would be well telegraphed, so people could react. Changing the default for how loans work to allow tokenholders to index themselves to inflation is an important part of this as well.

**Reply by Schwartz10** [reply]
@Fatman13 If the goal is to attract more capital to the ecosystem, why is preserving the original supply %'s of tokens an important quality to preserve? I've had many many conversations with various investors, and that has not ever come up in a single conversation. 

Also, obviously various stakeholder groups have bought/sold since the network launched, so the original allocation of tokens is already skewed and not preserved.

**Reply by Fatman13** [reply]
> Your argument seems to be that we should confiscate someone else's earned portion

That is not my argument. There are many crypto projects adopts burning mechanism to systematically burn what you refer to as "someone else's earned portion" like [BNB](https://cointelegraph.com/news/what-is-bnb-auto-burn-and-how-does-it-work) auto-burn which operates in good faith and does attract hype.  

> it isn't even definitively affecting the final distribution

I am confused. If the expectation is that the network would never surpass XX Eib, why burn it in the first place? 

> but it is affecting an incentive to put capital / resources into the network today

In your proposal...

> - Given there is still inflation slated at least until 2040 via simple minting, this proposal should not change the tokenholder incentives to lend or not lend
> - If anything, long term aligned tokenholders now have an juiced incentive for continuing to take long duration risk on FIL by continuing to lend and support SP growth during down markets

First you mentioned that the incentive is not changed, then you mentioned long-term holders are more incentivized. Are you suggesting there are more long-term holders so that the eventual outcome is that people are incentivized to invest?

> Burning the mining reserve / fundraising allocation doesn't actually solve for the time dimension that I laid out above.

Is the following "time dimension" you referring to? 

> It's important to contextualize this in terms of opportunity cost - if we do not add a time dimension to baseline minting, the rational choice is to deploy resources elsewhere (short term XRP trade, or putting capital against the Bitcoin halvening) because future rewards are always on the table.

Wouldn't burning mining reserve and fundraising allocation remainder be a much direct way to compete resources with XRP trading and Bitcoin halvening?

> the flag I'd received was that this may trigger larger debates given FIPing out things is a 1 way door vs a 2 way door.

Where could the flags come from? I don't expect flag raised from miners and ecosystem in general. IMHO, I see all FIPs are 1 way door. If the only concerns are larger debate and quicker shipping, i would suggest a more thorough fix than a compromised halfway fix.

>  it would allow folks to credibly point out what the total supply in 2040 would be on the network's current trajectory

IMHO, i highly doubt most people are looking that far into the future given the majority of the demand is from mining (buying pledge)...

**Reply by Fatman13** [reply]
> If the goal is to attract more capital to the ecosystem, why is preserving the original supply %'s of tokens an important quality to preserve?

The point is not about preserving %. AFAIU, the proposal come down to burning tokens and it chose miner allocation to burn from to achieve the said goal. What i am saying is that if the network collectively decide to burn then the burn should be fair. Two ways (could be many more) i see how it could be done fairly is that each parties in the FIL distribution burn their shares OR just burn the mining reserve and fund raising remainder.  

IMHO, the mining reserve and fundraising remainder to me is a much bigger uncertainty to investors than baseline.

**Reply by jnthnvctr** [reply]
> That is not my argument. There are many crypto projects adopts burning mechanism to systematically burn what you refer to as "someone else's earned portion" like [BNB](https://cointelegraph.com/news/what-is-bnb-auto-burn-and-how-does-it-work) auto-burn which operates in good faith and does attract hype.

Look if PL and FF want to do that, I'm all for it. I just don't think that's a FIP. 

> I am confused. If the expectation is that the network would never surpass XX Eib, why burn it in the first place?

One can have the view the network will grow (at some point), and if you hold that view - then this proposal doesn't change anything (or simply burns some small portion of unvested rewards until we're back to growth).
 
If you have the view the network will never exceed a specific size - then why play this game of having lots of deferred inflation (that discounts current FIL?)

> First you mentioned that the incentive is not changed, then you mentioned long-term holders are more incentivized. Are you suggesting there are more long-term holders so that the eventual outcome is that people are incentivized to invest?

Maybe I wrote it poorly, but what I meant is: the incentive to lend (yield) exists - that incentive does not go away. Doing nothing (sitting on them) vs doing something (lending your tokens) still comes at a cost because other inflation exists - at a minimum via simple minting. What changes is perhaps the _urgency_ given the amount of inflation decays, and the share of the pie it represents increases if future dilution is removed. 

> Wouldn't burning mining reserve and fundraising allocation remainder be a much direct way to compete resources with XRP trading and Bitcoin halvening?

Those are one offs though? I don't think the goal here is to aim for simply a one off - but create a better structural design. 

> Where could the flags come from? I don't expect flag raised from miners and ecosystem in general. IMHO, I see all FIPs are 1 way door. If the only concerns are larger debate and quicker shipping, i would suggest a more thorough fix than a compromised halfway fix.

I mean literally it was a flag that when I circulated to some core devs - the idea of all or nothing. Your comments though made me realize maybe a better path is to instead burn half or 2/3rds  - less future dilution; but still reserves some to allay concerns that some mining incentives may be needed in the future.

I think that that's still independent of also creating more structural fixes (which I'd argue the proposed tweaks to baseline minting address). While the baseline is declining - one could credibly say that at a specific trajectory the expected FIL in 2040 is XX (which given size of deferred baseline rewards would be substantial). Adding in some amt of the other reserves burn you mention would further strengthen.

> IMHO, i highly doubt most people are looking that far into the future given the majority of the demand is from mining (buying pledge)...

I mean both @Schwartz10 and I can share a list of tweets / convos we've had - but another good sanity check is this [poll](https://x.com/iamnotnicola/status/1736068418678256031?s=20) @nicola ran (honestly not sure if it was related to this but well timed). Even if its directionally right, I think its safe to say for a substantial portion of web3 investors FDV is a relevant thing to worry about

**Reply by protocolin** [reply]
Just to chime in: burning the fundraising remainder would be non trivially complicated as part of it has been allocated to ecosystem participants (and is intended to continue to support ecosystem participants separate from PL). The fundraising remainder was originally earmarked for ecosystem development and other activities, and PL has been allocating those tokens for activities like Filecoin ecosystem development.  This includes things like grants for developing alternate implementations, optimizing performance and cost of Filecoin proofs, development of wallet infrastructure, developer education and onboarding (eg. Hackathons, developer tools), client use case development, and much much more.

**Reply by kernelogic** [reply]
PL / FF portion needs to reduce along with baseline. 

With all these employees let go, the operational cost should reduce, but the PL / FF portion remains the same - makes no sense.

As an ecosystem developer, I am having a hard time to receive promised milestone payouts. The tokens are definitely not going to ecosystems now.

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
excuse me? 

the "fundraising remainder" (if i get you right here - that means the FIL pre-printed and still locked, right?) is "owned" by the protocol. the protocol itself is the source of it and the protocol can use them as it pleases. The Lord giveth and the Lord taketh away. 

this proposal makes it clear that the protocol, via FIP, should be able to change the originally proposed distribution rules of how we print FIL for cryptoeconomic reasons. there is no reasoning why this should't be applied to the, still locked, pre-printed FIL if the protocol decides to do so.

**Reply by Fatman13** [reply]
> Look if PL and FF want to do that, I'm all for it. I just don't think that's a FIP.

We can make it technical just like how [BEP-95](https://academy.binance.com/en/glossary/bep-95) was implemented for BNB. Basically introduce a new built in actor in the core protocol to [throttle](https://github.com/filecoin-project/FIPs/discussions/887#discussioncomment-7903745) / burn the multi-sig vesting addresses. Community stakeholder can all decide if the FIP is justified. 

> If you have the view the network will never exceed a specific size - then why play this game of having lots of deferred inflation (that discounts current FIL?)

I am with you on...

> On reflection, I'm actually not sure a deferred inflation model ever really made sense.

But given filecoin is born with baseline, IMHO, new changes made are to be fair to all parties at least to the initial design of FIL distribution. 

> Doing nothing (sitting on them) vs doing something (lending your tokens) still comes at a cost because other inflation exists

I think right now doing something (lending your tokens) doesn't generate enough interest for most to justify the loss of liquidity especially when people are comparing competing options. Do nothing can at least give you an advantage during FUD. Maybe someone could do an updated "where is the FIL" to check current market sentiment?

 > but create a better structural design.

Yes.

> I mean literally it was a flag that when I circulated to some core devs - the idea of all or nothing.

Most core devs expressed their [intent on being solely technical advisors](https://github.com/filecoin-project/core-devs/discussions/121). Their opinions on non technical issues matters not. The network stakeholder should collectively decide. IMHO, i don't see it as an all or nothing attempt. I see it as a good gesture towards community to restore faith in the project. 

> but still reserves some to allay concerns that some mining incentives may be needed in the future.

Colin's [comment](https://github.com/filecoin-project/FIPs/discussions/887#discussioncomment-7903564) says it all. There is no guarantees that the funds will be put into this use. There is no accountability. 

> one could credibly say that at a specific trajectory the expected FIL in 2040 is XX

Again, IMHO, mining reserve and fundraising are a much bigger uncertainty to investors than baseline.

> I mean both @Schwartz10 and I can share a list of tweets / convos we've had

What would be your estimation of FDV investors compared with institution, retail and mining investors size wise?

**Reply by Fatman13** [reply]
> Just to chime in: burning the fundraising remainder would be non trivially complicated as part of it has been allocated to ecosystem participants (and is intended to continue to support ecosystem participants separate from PL). The fundraising remainder was originally earmarked for ecosystem development and other activities, and PL has been allocating those tokens for activities like Filecoin ecosystem development. This includes things like grants for developing alternate implementations, optimizing performance and cost of Filecoin proofs, development of wallet infrastructure, developer education and onboarding (eg. Hackathons, developer tools), client use case development, and much much more.

IMHO, this may be the exact reason that we have to burn the mining reserve...

**Reply by jnthnvctr** [reply]
> PL / FF portion needs to reduce along with baseline. With all these employees let go, the operational cost should reduce, but the PL / FF portion remains the same - makes no sense.

I'm not sure how this makes sense - PL had way fewer employees when the network launched, we didn't increase their allocation as they hired more people. 

> this proposal makes it clear that the protocol, via FIP, should be able to change the originally proposed distribution rules of how we print FIL for cryptoeconomic reasons. there is no reasoning why this should't be applied to the, still locked, pre-printed FIL if the protocol decides to do so.

It isn't changing the original distribution, if we were to grow against baseline nothing would change.

> We can make it technical just like how [BEP-95](https://academy.binance.com/en/glossary/bep-95) was implemented for BNB. Basically introduce a new built in actor in the core protocol to https://github.com/filecoin-project/FIPs/discussions/887#discussioncomment-7903745 / burn the multi-sig vesting addresses. Community stakeholder can all decide if the FIP is justified.

Ack - I mean I agree, but imho if its a FIP / enforced against their will, it is confiscation of wealth.

> But given filecoin is born with baseline, IMHO, new changes made are to be fair to all parties at least to the initial design of FIL distribution.

I hear you, the point I'm making is it isn't changing the initial distribution (it is potentially changing a "final" distribution - in that it makes the network's success path dependent). But in either full growth or full shrink cases - you should have the same circulating supply at the end state.

> The network stakeholder should collectively decide. IMHO, i don't see it as an all or nothing attempt. I see it as a good gesture towards community to restore faith in the project.

I agree! It's why they were among a group of folks I'd first floated this by (across all groups, SPs, tokenholders, builders, core devs), to get early feedback and then put up a proposal.

I don't think its bad though to balance that with a small amt of humility that we don't know what we don't know - in my view there are two-ish potential solutions to that. 
(1) If we burn say 2/3 of the mining reserve - we get most of the value from burning it all, with a small exit door in case we need it in the future.
(2) Change the mining reserve so that it has no owner (TIL it had one) and which then requires a network upgrade / fork to re-assign an owner to actually spend if a FIP is ever introduced (so the trust is back at the protocol level). I had assumed this was always how it worked, but someone dm'd me to show me the code.

> Again, IMHO, mining reserve and fundraising are a much bigger uncertainty to investors than baseline.

Yes and no. I think FDV in general is (so inclusive of all sources) - which baseline is a larger contributor than the other two by kind of a large margin. That isn't to say we shouldn't also do something (e.g. the thing I suggested above for mining reserve) - but I think something with baseline should also be on the table if we want to have a long term effect. 

I want to be clear though - that its not that all inflation is bad - releasing rewards if the network is growing is good. From the tokenholder perspective, you just need avenues to capture a share of that inflation if you want to be able to "save" in FIL. This is why the proposal isn't to outright burn everything, but to keep the hw growth incentives (and separately from this proposal) improve how lending works ore broadly.

But I really do think the separate point about time urgency matters (i.e. why deploy in FIL today vs other places) and signaling matters too.

> What would be your estimation of FDV investors compared with institution, retail and mining investors size wise?

- Mining I don't know if folks care as much (since they capture inflation as it rises). This is complicated because depending on time horizon and operation - the thing you might care most about is just hedging to lock in gains. 

- Tokenholders in general care the most - retail and institutional I think are kind of the same here; though I think the history of the Sam coins is a tax we sort of pay. A pretty standard thing when institutional folks will take FIL to IC is trying to understand the inflation dynamics and how to project out the FDV theyre underwriting on a N year timeframe (1-3 as an example). It really isnt great to have to include probability distributions etc of what constellation of miner dynamics might put you at at various points of the inflation spectrum at some future point in time. At the end of the day (as Jschwartz mentioned too) - if its too complicated to underwrite they just focus elsewhere (attention is scarce!).

**Reply by Fatman13** [reply]
> Ack - I mean I agree, but imho if its a FIP / enforced against their will, it is confiscation of wealth.

I think phrasing a common practice (like BEP-95) within crypto communities in a way that entails negative connotation may suggest to me that the thinking behind is not as crypto. This is a Crypto / Blockchain community after all.

> Change the mining reserve so that it has no owner (TIL it had one) and which then requires a network upgrade / fork to re-assign an owner to actually spend if a FIP is ever introduced (so the trust is back at the protocol level).

I think that may still present a major threat to investors given how FIP is governed right now and with FIP0001v2 suggesting a totalitarian design that a group of 7 people may [control the network](https://github.com/filecoin-project/FIPs/discussions/799#discussioncomment-7842643) which is btw in complete violation of [filecoin mission](https://github.com/filecoin-project/FIPs/blob/master/mission.md). At this point, IMHO, burning the mining reserve and the fund raising remainder seem to be the most straight forward way to encourage capital. 

> From the tokenholder perspective, you just need avenues to capture a share of that inflation if you want to be able to "save" in FIL.

I think "avenues" in this context may not be FIL at all (as you pointed out in the next paragraph). It can be BTC or ETH. Investors look for "avenues" to hedge against inflation with whatever they think is the best asset class for them and it doesn't necessarily have to be FIL assets. Again it would be best if we do an updated version of "Where is the FIL". But, IMHO, by eliminating threats like mining reserve and the fund raising remainder, we would have much better chance that investors see FIL as an "avenue".

> (i.e. why deploy in FIL today vs other places)

+1

> if its too complicated to underwrite they just focus elsewhere (attention is scarce!)

+1 to scarce attention. I still think burning mining reserve and the fund raising remainder is the most straight forward way to achieve the stated goals as those funds don't even need to dig how baseline works and it is fair to most participants of the network.

**Comment by wheresaddie** [comment]
LFG

**Comment by jennijuju** [comment]
Thank YOU for not only seeing the problem (inflation), but also raises potential solutions! Especially taking application developers (like Defi) into consideration and how they could play a role on filecoin token economics as they were nonexistent when network CE was firstly designed!

**Comment by AxCortesCubero** [comment]
I have a few concerns with this idea, the first one being the motivation. The solution to "Filecoin’s tokenomics are broadly misunderstood" is not to make a quite large cryptoeconomic change to make it easier to understand, the most direct answer would be making sure we can explain it better. The baseline minting mechanism is somewhat intricate but i'm sure we can find ways to explain this message at various level of detail.

The baseline minting mechanism was established for a reason that I think still makes sense, or I haven't been convinced it is no longer applicable. I see two main motivations for having this mechanism: first is as you say, saving for the future, make sure we don't overincentivize early adopters, if rewards decay exponentially too quickly, we might burn through them before we have an established sustainable economy where clients pay for storage services and this is enough incentive. The second reason is that when we are below baseline, the minting mechanism encourages collaborative competition, the more power is onboarded the more total rewards are given out, which is good for network growth, opposed to the pure competition that happens above baseline. Baseline mechanism is not necessarily perfect though, and open for improvements (I think for instance it is unreasonable to set ourselves to compete against exponential growth forever).

I think Ethereum has harder tokenomics to explain than Filecoin in this regard, for investors concerned about FDV, where their token supply is unbounded, so those worried about the eventual value of their token in the infinite future, well their token will be diluted to exactly zero. Of course no one cares about the infinite future, you care more about money that is closer to the present, than the far future, what happens to supply in the next 10 years is much more important than the next 20 years and so on, you can get a better idea of what your token would be worth to you this way (again this is a nice blog post or communication that could be written to communicate with investors, rather than drastically changing our CE).

Another point is also, I sort of get your idea, but it is ill-defined. Baseline minting doesn't exactly work that way that we save what wasn't minted to give it out separately. It is unclear what it means that this difference between what was minted and what would have been minted would be burned. Those tokens we save now, are not given out at a particular time in the future, they are spread out and just increase the size of minting for all the future. So more properly, anything that is saved is used to change the shape of the baseline minting function forever, fatten the tail. You'd need to specify what you intend to do, you want to use this difference to thin out this tail of the baseline minting function based somehow on what wasn't minted. I get the spirit of what you want to say, but it is underspecified, this could happen many ways.

**Reply by Schwartz10** [reply]
> I have a few concerns with this idea, the first one being the motivation

I think the proposal could've separated out the "real" motivation from the approach to achieving the desired end result. Just to be clear from my standpoint, the motivation is to "attract more capital from external investors into the Filecoin ecosystem". From what I can tell, your objection to the proposal seems around the approach to achieving this desired outcome. Hopefully we can agree that attracting capital is an important motivation. 

> I have a few concerns with this idea, the first one being the motivation ... the most direct answer would be making sure we can explain it better ... I think Ethereum has harder tokenomics to explain than Filecoin in this regard

When I read these statements, i interpret this as you saying "the correct approach to bringing more capital into the ecosystem is better education". Through fundraising efforts for GLIF, i've had conversations with hundreds of different investors of various sizes. I've learned that investors are more than capable of understanding, but it's a constant uphill battle that plagues the network for no immediately productive use (more on this in my next point). 

- Investors can take a long time to understand, slowing down the process of teams within the Filecoin ecosystem attracting capital to build. In that time, teams can lose motivation, focus, or get forced to pivot and build in an ecosystem that has more funding opportunities
- Investors do not have infinite time to learn about Filecoin, so a lot of them won't write it off, but they just won't take the time to understand. They'll just follow others (sheep problem). 
- Part of the problem is how block explorers track and report on tokenomics, which we have no control over.
- Arguing that "ETH tokenomics is more complicated than FIL tokenomics" is not a productive argument because ETH is an entirely different asset with far more maturity and proof points of sustainable deflationary periods. Investors thinking about underwriting FIL  are doing different types of diligence than they are when underwriting ETH. 

> Of course no one cares about the infinite future, you care more about money that is closer to the present, than the far future, what happens to supply in the next 10 years is much more important than the next 20 years and so on

I completely agree with this, which is why it's hard for me to understand your objection to the proposal. In case it wasn't clear, the proposal should not change _anything_ until baseline _would_ have run out (which I believe would be scheduled for 2040 if the network had perpetually stayed above baseline). Saying "baseline is still correct because of ... saving for the future", you're hurting Filecoin now for a (maybe better?) outcome in 2040. This is like investing in saddles in your horse and carriage business in the early 1900s - you simply can't know what the state of crypto markets and filecoin will be in 2040 to say baseline is "still correct today". From all the conversations I've personally had, baseline is not correct, filecoin should be more aggressively burnt. 

> make sure we don't overincentivize early adopters, if rewards decay exponentially too quickly, we might burn through them before we have an established sustainable economy where clients pay for storage services and this is enough incentive

The proposal would change _none_ of these points.

**Reply by AxCortesCubero** [reply]
> Hopefully we can agree that attracting capital is an important motivation.

Yes, I agree that attracting capital is important, I just don't agree that making a huge cryptoeconomic change like this is the best way to attract capital.

The argument is that this change would make the numbers seem more reasonable for present investors, because it would improve their FDV. The cost of doing this change is that cutting out future incentives that could help Filecoin have a more sustainable future (we don't know when Filecoin will become self sustainable, where SP's earn sufficient rewards from paid deals rather than block rewards). So the risk is that this might be a short sighted plan for the longevity of Filecoin: we may improve somewhat our present fundraising, at the cost of future stability.

A simpler version of this proposal would be: both simple minting and baseline minting are based on exponential decay with a 6 year half life. We could propose to shorten this to 3 years or whatever, so block rewards decay faster. This would improve FDV, but puts more pressure on Filecoin becoming self-sustainable sooner, since block rewards would deplete faster. 


It comes down to understanding the potential benefit: how much better would FDV-like calculations look for investors, vs how much the downside of how much pressure we are putting on future Filecoin (Which depends how much we care about future Filecoin, if we just want it to survive for the next 10 years, and then whatever, then sure this is a good idea).

I'd argue we'd still need to communicate to potential investors the idea of time-weighting, as I explained for ETH where FDV doesn't make sense.  Let's say we can present them a time-weighted FDV (twFDV). This is something we'd have to do some math on, but what I think is that the twFDV would not see much improvement from this proposal. This is because this proposal isn't changing much in the near future for potential investors, and the near future receives higher weight. Once we start seeing the effects of this proposal, this is already weighted down and less important for computing the twFDV.

I understand investors don't have a lot of time to spare on understanding Filecoin, but I think something like twFDV is something that can be presented and understood fairly quickly.

> Saying "baseline is still correct because of ... saving for the future", you're hurting Filecoin now for a (maybe better?) outcome in 2040.

So again my point is the benefit you'd see in terms of twFDV for investors, will probably be small compared to the the sacrifice we are passing on to future Filecoin.

> In case it wasn't clear, the proposal should not change anything until baseline would have run out (which I believe would be scheduled for 2040 if the network had perpetually stayed above baseline).

I don't really understand what you mean here. Baseline minting is not something that runs out, it just becomes smaller and smaller with time, but there is not a particular date when we are scheduled to deplete it (it doesn't go stepwise say like Bitcoin halving)


> make sure we don't overincentivize early adopters, if rewards decay exponentially too quickly, we might burn through them before we have an established sustainable economy where clients pay for storage services and this is enough incentive

>The proposal would change none of these points.

I mean, it does change that, I don't understand why you say it doesnt, that is the point of the proposal, to deplete block rewards faster, you do this by removing rewards that would have gone to future SP's. Present SP's would now be overincentivized compared to future SP's, at least relative to the current reward system we have.  So you are trying to make FDV (or twFDV) more attractive to investors at the cost of future SP incentives.

**Comment by jimmylee** [comment]
This is a great idea

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
you have some math on this? theoretically we still have 5.4EiB to loose to the x10 nonsense. i see a clear incentive to lower raw storage capacity constantly to stay below the baseline as much as possible from the miner side in this proposal

i see a immediate net outflow on the hardware side if this gets implemented.

**Reply by jnthnvctr** [reply]
If simple minting is going, and the same baseline mechanic for growth exists (growing hw adds rewards) - why would we see a net outflow of hardware?

For better or worse, this doesn't affect the 10x trend. But if baseline shrinks and we start destroying future FIL, it makes the current FIL a higher share of a shrinking pie - should that not delay hw coming offline?

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
because you want to put an incentive on staying under the baseline. if the proposal is beneficial, as you claim, then why shouldn't the biggest group of fil holders - miners - decide to exploit the burn route. if they can, example numbers here, print one and burn 2 they might decide that that is more beneficial to them than going at print 2 burn 1 or the original print 1.5 and 1.5 in 349568 years. 

x10 is just a nice way to keep the baseline down while maintaining investment in FIL and reducing hardware from a miner side.

**Comment by EVS2024** [comment]
Just to be clear, if this proposal goes into effect after the next upgrade, the equivalent QAP will immediately get lower rewards than it would have without this proposal?

**Reply by Schwartz10** [reply]
No, there should not be any changes until after baseline _would have_ run out given RBP > baseline for the entire lifetime of filecoin. 

To summarize - nothing changes about rewards until 2040

**Comment by deltazxm** [comment]
We can burning Mining Reserve first.

**Comment by AxCortesCubero** [comment]
I have given some thought to how such "use it or lose it" mechanisms could look like. As I understand it, there are two possible ways to prevent baseline minting from reaching its maximum asymptotic value in the far future:

1) reduce the minting rate, such that a different asymptotic value is reached,
or
2) Keep the same minting rate, but stop minting at some point, when we have minted enough

I elaborate some more what these options mean and could look like in [this doc](https://hackmd.io/@cryptoecon/SkIdosSF6), but they both come with some drawbacks to think about.

In option 1) It is not true that this only affects future Filecoin minting, and wouldn't affect us today. The minting rate reduction would be felt immediately, so it is not just problems for the future. On top of that there is a possible feedback loop that could make Baseline minting rewards quickly drop to zero, this works like: Because we had less minting, this means we will mint even less, which means we will mint even less...

With option 2) the main risk is that At some point there will be a huge shock to the network, when the baseline minting just suddenly stops. We'd need some more analysis to have an idea when exactly that shock would happen and how large it would be.

**Comment by aarshkshah1992** [comment]
@jennijuju @rjan90 @jnthnvctr What's the best way to unblock this and proceed with a governance vote here to see if majority of our community is aligned with this or not ?

**Reply by jennijuju** [reply]
@aarshkshah1992 As discussed earlier, I think the next step is to work on concrete proposal that's coming out of this dicussion. If you have ideas, I'd encourage you to work with @jnthnvctr @AxCortesCubero tgt!
